### DISCUSSION TITLE
Example compilation error for UQTk on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by ritwika-chakraborty** [sandialabs/UQTk#12] [Q&A] [answered]
Hello, I am new to using UQTk. I have successfully installed UQTk 3.1.4, but I am having trouble compiling and running the examples. For instance, I followed the manual to compile and execute /examples/ops/ and get the following error with make command: 
`[  2%] Building CXX object ops/CMakeFiles/Ops.x.dir/Ops.o
In file included from /home/ritwika/UQTk/examples/ops/../../cpp/lib/pce/PCSet.h:43,
                 from /home/ritwika/UQTk/examples/ops/Ops.cpp:30:
/home/ritwika/UQTk/examples/ops/../../cpp/lib/include/ftndefs.h:75:4: error: #error "FTN_NAME macros not defined for this system"
   75 |   #error "FTN_NAME macros not defined for this system"
      |    ^~~~~
In file included from /home/ritwika/UQTk/examples/ops/Ops.cpp:30:
/home/ritwika/UQTk/examples/ops/../../cpp/lib/pce/PCSet.h:47:10: fatal error: cvode/cvode.h: No such file or directory
   47 | #include <cvode/cvode.h>             /* prototypes for CVODE fcts., consts.  */
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [ops/CMakeFiles/Ops.x.dir/build.make:76: ops/CMakeFiles/Ops.x.dir/Ops.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:468: ops/CMakeFiles/Ops.x.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`
It seems to me that there was a problem with gfortran, but I did not encounter any issues during installation and I verified that I have gfortran 11.4.0 installed in my system. Is there any obvious mistake that I am making?

**Comment by bjdebus** [comment]
Hi @ritwika-chakraborty , thanks for reaching out. It looks like some macros have not been initialized properly. I can look into that. However, before I do that, can you verify for me that you did do the full build before trying to work with the examples? Did you run the CMake configuration script and then build the code with make? If that finished properly, you should be able to run ctest to verify that the build went well. If none of this makes sense, let me know and I can be more specific in my instructions. 
Thanks again,
Bert

**Comment by ritwika-chakraborty** [comment]
Thank you for the quick response @bjdebus . Yes, I have successfully built the code and run ctest. It passes 100% of the tests:
![Screenshot from 2024-04-12 21-46-54](https://github.com/sandialabs/UQTk/assets/119865625/7ea7b66c-1939-46a3-9025-bf4c9ffee6ed)
Ritwika

**Comment by bjdebus** [comment]
Ok, if the build was successful, and you did a "make install", then all examples in the install folder should already be compiled. Are you recompiling because you were making changes to the examples? Or did something go wrong with the install?

**Comment by ritwika-chakraborty** [comment, marked as answer]
Very sorry for being vague. The problem was I was trying to run the examples in the source directory instead of the install directory. Everything works fine now. I can run and compile all the examples. Thank you for all your help. :)

**Comment by bjdebus** [comment]
Glad to hear you figured it out. Feel free to reach out if you run into any other issues!
Bert
